server-timing/defaults.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * Server-Timing API default metrics
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Registers the default Server-Timing metrics for before rendering the template.
+ *
+ * These metrics should be registered as soon as possible.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_default_server_timing_before_template_metrics() {
+	$calculate_before_template_metrics = function() {
+		// WordPress execution prior to serving the template.
+		perflab_server_timing_register_metric(
+			'before-template',
+			array(
+				'measure_callback' => function( $metric ) {
+					// The 'timestart' global is set right at the beginning of WordPress execution.
+					$metric->set_value( ( microtime( true ) - $GLOBALS['timestart'] ) * 1000.0 );
```

Comment:
is `$GLOBALS['timestart']` always defined? if not, add a check here

Reply:
@adamsilverstein Yes it will always be set, since WP sets it unconditionally even before the database is initialized.

The only potential edge-case would be if a plugin simply unset the global, but that would be a problem in that plugin. I'm undecided whether it's worth adding a check for.

Reply:
probably safe to assume it is available, I don't see any plugins modifying it and query monitor uses it directly: https://wpdirectory.net/search/01GM8ZBD0ZT8XF4YGXKHA06XHK

Reply:
Thanks for the extra research! I had no idea this website existed, super useful 😍